### ISSUE TITLE
config.get(section, opt) doesnt coerce value to appropriate type 

### ISSUE THREAD

**Issue by gurobokum** [iterative/dvc#3154]
`config.get(section, opt)` returns value as str when expects int

### Steps for reproduce
1. Add number value into config section
   ```
   [core]
   checksum_jobs = 10
   ```
   ```
   SECTION_CORE_CHECKSUM_JOBS: All(Coerce(int), Range(1))
   ```
2. Run cli
3. config.get returns str
   ```
   >>> from dvc.repo import Repo
   >>> r = Repo()
   >>> r.config.get('core', 'checksum_jobs')
   '10' # but expects to get 10 as int
   >>> r.config.config.get('core', {}).get('checksum_jobs')
   10
   ```
**Please provide information about your setup**
```
0.81.3+aec778
Linux (pip)
```

**Comment by mroutis** [comment]
@JIoJIaJIu, this is because `configobject` is a wrapper of `configparser`.

`ConfigObject` has methods like `.as_int()` to work around this (unnecessary, in my opinion).
I would rethink the current `Config` implementation.

**Comment by Suor** [comment]
The design of our `Config` object is not clear and simple) We discussed refactoring it with @efiop, but it still waiting and we are working around things like that for now.

**Comment by skshetry** [comment]
Closed by #3298.
